File: v1/ruby/AuthToken_Test.rb
require "test/unit"
require_relative './AuthToken'

# Assumption:
# Server running in localhost:9080 with user-password `admin`/`admin`.
class TestClient < Test::Unit::TestCase

  def test_negative
    client = StriimRestClient.new('http://localhost:9080')
    assert_raise(RuntimeError) { client.auth_token('admin', 'test') }
  end

  def test_positive
    client = StriimRestClient.new('http://localhost:9080')
    assert_equal(36, client.auth_token('admin', 'admin').length)
  end
              
end

File: v1/python/AuthToken_Test.py
import unittest
from AuthToken import AuthToken

class TestAuthTokens(unittest.TestCase):

    def test_positive(self):
       authtoken = AuthToken()
       self.assertIsNotNone( authtoken.authenticate('admin','admin'),'Authenticated Successfully')

    def test_negative(self):
       authtoken = AuthToken()
       self.assertIsNone( authtoken.authenticate('admin','test'),'Authentication Failure')

if __name__ == '__main__':
     unittest.main()
